src/context/ast/QueryAstContext.h

```diff
@@ -1,188 +1,59 @@
-    std::vector<std::string>                          projOutputColumnNames_;
-};
-
-struct ReturnClauseContext final : CypherClauseContextBase {
-    ReturnClauseContext() : CypherClauseContextBase(CypherClauseKind::kReturn) {}
-
-    std::unique_ptr<OrderByClauseContext>        order;
-    std::unique_ptr<PaginationContext>           pagination;
-    std::unique_ptr<YieldClauseContext>          yield;
-};
-
-struct WithClauseContext final : CypherClauseContextBase {
-    WithClauseContext() : CypherClauseContextBase(CypherClauseKind::kWith) {}
-
-    std::unique_ptr<OrderByClauseContext>       order;
-    std::unique_ptr<PaginationContext>          pagination;
-    std::unique_ptr<WhereClauseContext>         where;
-    std::unique_ptr<YieldClauseContext>         yield;
-    std::unordered_map<std::string, AliasType>  aliasesGenerated;
-};
-
-struct MatchClauseContext final : CypherClauseContextBase {
-    MatchClauseContext() : CypherClauseContextBase(CypherClauseKind::kMatch) {}
-
-    std::vector<NodeInfo>                       nodeInfos;
-    std::vector<EdgeInfo>                       edgeInfos;
-    std::unique_ptr<PathBuildExpression>        pathBuild;
-    std::unique_ptr<WhereClauseContext>         where;
-    std::unordered_map<std::string, AliasType>* aliasesUsed{nullptr};
-    std::unordered_map<std::string, AliasType>  aliasesGenerated;
-};
-
-struct UnwindClauseContext final : CypherClauseContextBase {
-    UnwindClauseContext() : CypherClauseContextBase(CypherClauseKind::kUnwind) {}
-
-    const YieldColumns*                         yieldColumns{nullptr};
-    std::unordered_map<std::string, AliasType>* aliasesUsed{nullptr};
-    std::unordered_map<std::string, AliasType>  aliasesGenerated;
+enum FromType {
+    kInstant,
+    kVariable,
+    kPipe;
 };
 
-struct MatchAstContext final : AstContext {
-    // Alternative of Match/Unwind/With and ends with Return.
-    std::vector<std::unique_ptr<CypherClauseContextBase>>  clauses;
+struct Starts {
+    FromType fromType{kInstant};
+    Expression* src{nullptr};
+    Expression* originalSrc{nullptr};
+    std::string userDefinedVarName;
+    std::string firstBeginningSrcVidColName;
+    std::vector<Value> vids;
 };
 
-struct PatternContext {
-    PatternContext(PatternKind k, MatchClauseContext* m) : kind(k), matchClauseCtx(m) {}
-    const PatternKind kind;
-    MatchClauseContext*    matchClauseCtx{nullptr};
+struct Over {
+    bool isOverAll{false};
+    std::vector<EdgeType> edgeTypes;
+    storage::cpp2::EdgeDirection direction;
+    std::vector<std::string> allEdges;
 };
 
-struct NodeContext final : PatternContext {
-    NodeContext(MatchClauseContext* m, NodeInfo* i)
-        : PatternContext(PatternKind::kNode, m), info(i) {}
+// path context
+struct PathContext {
+    Starts from;
+    Starts to;
+    Steps steps;
+    Over over;
 
-    NodeInfo*            info{nullptr};
+    bool isShortest_{false};
```

Comment:
The format `label_` is used for private field.

Reply:
will fix in next pr